src/common/snippets/src/pass/fc_tokenization.cpp

```diff
@@ -0,0 +1,33 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "snippets/pass/fc_tokenization.hpp"
+
+#include "openvino/pass/graph_rewrite.hpp"
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+#include "snippets/itt.hpp"
+#include "snippets/op/subgraph.hpp"
+#include "snippets/utils/tokenization_utils.hpp"
+
+ov::snippets::pass::TokenizeFCSnippets::TokenizeFCSnippets(const SnippetsTokenization::Config& config) {
+    MATCHER_SCOPE(TokenizeFCSnippets);
+    // TODO: extend constant path coverage:
+    // 1. Transpose support
+    // 2. Convert support
+    // 3. Decompression subgraphs support (and all the possible compressed weights related precisions)
+    auto constant = ov::pass::pattern::wrap_type<ov::op::v0::Constant>();
+    auto m_matmul = ov::pass::pattern::wrap_type<ov::opset1::MatMul>({ov::pass::pattern::any_input(), constant});
+
+    auto callback = [OV_CAPTURE_CPY_AND_THIS](ov::pass::pattern::Matcher &m) {
+        OV_ITT_SCOPED_TASK(ov::pass::itt::domains::SnippetsTransform, "Snippets::op::TokenizeFCSnippets")
+        const auto matmul = m.get_match_root();
+        if (transformation_callback(matmul)) {
+            return false;
+        }
+        return ov::snippets::utils::tokenize_node(matmul, config);
```

Comment:
Do we really need to extract the common tokenization logic to utilities just to tokenize one op?
Can we use the following code here?
```
        auto subgraph = op::Subgraph::wrap_node_as_subgraph(matmul);
        subgraph->get_rt_info()["originalLayersNames"] = matmul->get_friendly_name();
        ov::replace_node(matmul, subgraph);
        op::update_out_tensor_name(subgraph);
```

Probably I missed something 🤔 Could you please elaborate your decision with `ov::snippets::utils::tokenize_node`?

The only one thought is to connected to the existing Subgraphs on first input 🤔

Reply:
> The only one thought is to connected to the existing Subgraphs on first input

You are right, that is one of the reasons why I reused the helper for FC tokenization. Thanks to that, we cover e.g. "Transposes on inputs + FC" cases by tests

 Also, the moving of tokenization logic in a separate helper is a good prerequisite for Tokenization refactoring. Currently, the tokenization looks as follows (I cut the code a bit):
```
    auto tokenization_passes = manager.register_pass<ov::pass::GraphRewrite>();
    tokenization_passes->add_matcher<TokenizeGNSnippets>();
    tokenization_passes->add_matcher<TokenizeFCSnippets>(m_config);
    tokenization_passes->add_matcher<TokenizeSnippets>(m_config);
```
`TokenizeSnippets` here is responsible for tokenization of the ops which were not tokenized by the previous passes. The pass contain a lot of checks, which are related to different ops, in one place (`is_supported_op` lambda): as a result, the pass looks quite cumbersome.

Ideally, I would want to separate this large pass into sequence of small matcher passes, which will match only on the specific ops, and contain only the checks which are needed for the matched ops:

```
    auto tokenization_passes = manager.register_pass<ov::pass::GraphRewrite>();
    tokenization_passes->add_matcher<TokenizeGNSnippets>();
    tokenization_passes->add_matcher<TokenizeFCSnippets>(m_config);
    tokenization_passes->add_matcher<TokenizeUnaryEltwise>(m_config);
    tokenization_passes->add_matcher<TokenizeBinaryEltwise>(m_config);
    tokenization_passes->add_matcher<TokenizeTranspose>(m_config);
    tokenization_passes->add_matcher<TokenizeMatmul>(m_config);
    ...
```
I believe this will improve tokenization code readability, and could allow us to configure the pipeline more precisely (e.g. `TokenizeTranspose` and `TokenizeMatmul` are used only in tests, so they can be disabled by the plugin in product code).

And each of this new matchers can reuse `ov::snippets::utils::tokenize_node` -- that's another reason why I moved the tokenization code in a separate helper.

Reply:
> You are right, that is one of the reasons why I reused the helper for FC tokenization. Thanks to that, we cover e.g. "Transposes on inputs + FC" cases by tests

Cool! Thank you for the explanation! I just wanted to check that I fully got the goal of this PR :blush:
Yeah, I saw some tests like `FC->Eltwise->FC`.

> Ideally, I would want to separate this large pass into sequence of small matcher passes, which will match only on the specific ops, and contain only the checks which are needed for the matched ops

This is really good idea! 🤔